src/init.cpp

```diff
@@ -975,6 +976,41 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
         fEnableReplacement = (std::find(vstrReplacementModes.begin(), vstrReplacementModes.end(), "fee") != vstrReplacementModes.end());
     }
 
+    if (!mapMultiArgs["-bip9params"].empty()) {
+        // Allow overriding bip9 parameters for testing
+        if (!Params().MineBlocksOnDemand()) {
```

Comment:
As mentioned in https://github.com/bitcoin/bitcoin/pull/8460/commits/f36be17578d46235a4bd63754dbba6e664d64117#r73609765 this could have anew parameter instead of reusing mineblocksOnDemand.
 
What about fAllowOverwriteBIP9Softforks?